src/transactions.rs

```diff
@@ -97,10 +100,13 @@ impl<L: Ledger> Transaction<L> {
     pub fn timeout(&self) -> Option<u64> {
         self.timeout
     }
+    pub fn hash(&self) -> &Option<TransactionHash<L>> {
```

Comment:
Nit: it's probably a tiny bit more conventional from an API perspective to return `self.hash.as_ref() : Option<&TransactionHash<L>>`. See e.g. [HashMap::get](https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html#method.get) or [Vec::last](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.last).